test/constraint.jl

```diff
@@ -1,3 +1,13 @@
+using JuMP
+using LinearAlgebra
+using Test
+
+include(joinpath(@__DIR__, "utilities.jl"))
+
+@static if !(:JuMPExtension in names(Main))
```

Comment:
Why do we do this unusual check for JuMPExtension.jl but not utilities.jl?

Reply:
Utilities just loads some functions, whereas JuMPExtension is a module so we get annoying `WARNING: module JuMPExtension has been redefined` warnings.

The other PRs fix this by making each file a module. The module approach also prevents state leaking between files, and allows us to programmatically determine the tests to run within a particular file.